docs/src/main/asciidoc/websockets-next-reference.adoc

```diff
@@ -998,6 +998,48 @@ public class ProductEndpoint {
 <1> The `getProduct` callback method can only be invoked if the current security identity has an `admin` role or the user is allowed to get the product detail.
 <2> The error handler is invoked in case of the authorization failure.
 
+==== Bearer token authentication
+
+The xref:security-oidc-bearer-token-authentication.adoc[OIDC Bearer token authentication] expects that the bearer token is passed in the `Authorization` header during the initial HTTP handshake.
+Some WebSocket clients like the https://vertx.io/docs/vertx-core/java/#_websockets_on_the_client[Vert.x WebSocketClient] support configuring headers to the WebSocket opening handshake, but clients that strictly follow the https://websockets.spec.whatwg.org/#the-websocket-interface[WebSockets API] do not support headers.
+The absence of headers proves a challenge when writing secure WebSocket client applications for a web browser with JavaScript.
+The JavaScript WebSocket client only allows to configure the HTTP `Sec-WebSocket-Protocol` request header.
+We have seen users to take advantage of this header for purpose which it is not intended, that is to propagate the `Authorization` header.
+Here is an example of a JavaScript client propagating the `Authorization` header as a subprotocol value:
+
+[source,javascript]
+----
+const token = getBearerToken()
+const quarkusHeaderProtocol = encodeURIComponent("quarkus-header#Authorization#Bearer " + token) <1>
+const socket = new WebSocket("wss://" + location.host + "/chat/" + username, ["irrelevant", quarkusHeaderProtocol]) <2>
+----
+<1> Expected format for the Quarkus Header subprotocol is `quarkus-header#header-name#header-value`.
+Do not forget to encode the subprotocol value as a URI component to avoid encoding issues.
+<2> Indicate 2 subprotocols supported by the client, the subprotocol of your choice and the Quarkus headers subprotocol.
+For purpose of the header propagation, it is not necessary to replace the `irrelevant` subprotocol value.
```

Comment:
```suggestion
```

IMHO it is simpler to give some technical name like `bearer-token-carrier` or something similar and note that the name of this sub-protocol is irrelevant for the purpose of transporting the custom headers

Reply:
I think the first sentences makes it clear that users choose the protocol of their choice, so this sentence is surplus to requirements and I'll drop it.